### ISSUE TITLE
Exempt HTML comments from indentation rule

### ISSUE THREAD

**Issue by jacobq** [ember-template-lint/ember-template-lint#514]
It would be nice if there were a way to exempt HTML comments (allowed via `"no-html-comments": false`) from the `block-indentation` rule so that neither of the comments in the example below would be considered violations. Otherwise if one turns off the rule disallowing HTML comments it becomes necessary to indent them like other elements even though it is often desirable to place comments at the root level for high conspicuity. 

```html
<section>
  <p>Lorem ipsum...</p>
  <!-- Ignore this line (at normal level) -->
</section>
<section>
  <p>Second example</p>
<!-- 
  Ignore all this too (at root level)
-->
</section>
```

**Comment by elwayman02** [comment]
This same issue exists for HBS comments. The block-indentation rule has no concept of how to treat comments, so if you were to comment out a component block, it would error:

```
{{!#foo-bar}}
  {{!stuff}}
{{!/foo-bar}}
```

The middle line errors because it's indented "incorrectly". This should at least be a configurable exception, if not default behavior.

**Comment by rwjblue** [comment]
Sorry I've missed this, I think ignoring handlebars comments definitely seems good (as a config option).

**Comment by lifeart** [comment]
Closing issue, but it's still listed here: https://github.com/ember-template-lint/ember-template-lint/issues/3209
